tests/test_parse_regression.py

```diff
@@ -0,0 +1,40 @@
+import filecmp
+import unittest
+import json
+from src.api import parse
+import os
+
+
+class TestParseRegression(unittest.TestCase):
+    def test_regression(self):
+        current_path = os.path.abspath(os.path.dirname(__file__))
+        testdir = current_path + "/testdata/"
+
+        config_files = [f for f in os.listdir(testdir) if f.endswith(".json")]
+
+        for config_file in config_files:
+            with open(testdir + config_file) as cfg:
+                config_raw = cfg.read()
+                config = json.loads(config_raw)
+
+            iocs = parse(config["type"], config)
+
+            result_file_txt = "test_result.txt"
+            with open(testdir + result_file_txt, "w") as fp:
```

Comment:
a bit weird - why save the test results to a temporary file (and not clean it up later).

Instead you should read a expected file to memory and just compare two variables (`assertEqual(expected_file_contents, iocs.prettyprint()`)